dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/src/Audacia.CodeAnalysis.Analyzers/Rules/ControllerActionReturnTypedResults/ControllerActionReturnTypedResultsInsteadOfIActionResultAnalyzer.cs

```diff
@@ -0,0 +1,91 @@
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class ControllerActionReturnTypedResultsInsteadOfIActionResultAnalyzer : DiagnosticAnalyzer
+    {
+        public const string Id = DiagnosticId.UseTypedResultsInsteadofIActionResult;
+
+        public const DiagnosticSeverity Severity = DiagnosticSeverity.Warning;
+
+        private const string MessageFormat = "Controller action name '{0}' should return a TypedResult rather than an IActionResult";
+
+        private const string Title = "Controller action should return a TypedResult rather than an IActionResult";
+
+        private const string Description = "Controller actions should return a TypedResult rather than an IActionResult.";
+
+        private const string Category = DiagnosticCategory.Maintainability;
+
+        private const bool IsEnabled = true;
+
+        private static readonly DiagnosticDescriptor Rule
+            = new DiagnosticDescriptor(Id, Title, MessageFormat, Category, Severity, IsEnabled, Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(Rule);
+
+        /// <summary>
+        /// A collection of syntax kinds that we want our analyzer to read.
+        /// </summary>
+        private readonly SyntaxKind[] _syntaxKinds =
+        {
+            SyntaxKind.MethodDeclaration
+        };
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(
+                analysisContext =>
+                {
+                    analysisContext.RegisterSyntaxNodeAction(
+                        AnalyzeMethodDeclaration,
+                        _syntaxKinds
+                    );
+                }
+            );
+
+        }
+
+        /// <summary>
+        /// The method declaration analysis includes the following checks:
+        /// 1. determines whether the method is a controller
+        /// 2. determines whether the controller has return type TypedResults instead of IActionResult
+        ///
+        /// A diagnostic will be reported if a method is a controller but the method has return type IActionResult instead of TypedResults.
+        /// Please note, this ONLY applies to controller actions.
+        /// </summary>
+        private static void AnalyzeMethodDeclaration(SyntaxNodeAnalysisContext nodeAnalysisContext)
+        {
+            var isController = nodeAnalysisContext.IsControllerAction();
+
+            if (isController)
+            {
+                var methodDeclarationSyntax = (MethodDeclarationSyntax)nodeAnalysisContext.Node;
+
+                var returnType = methodDeclarationSyntax.ReturnType.ToString();
+
+                if (returnType.Contains("IActionResult"))
```

Comment:
Probably less likely to be an issue, but can this check do something similar to the feedback for the TypesResults check, and do something like check the namespace?

Reply:
Done

Reply:
this doesn't seem to have been actioned to me?